homework-submission/week4/src/store/TodoListStore.js

```diff
@@ -0,0 +1,49 @@
+import { observable, action, computed, configure } from "mobx";
+import Data from "../TODOS.json";
+
+configure({ enforceActions: true })
+class TodoListStore {
+
+  @observable
+  TODOListItems = Data;
+
+  @action
+  checkStatus = (id) => {
+    const newTodoItems = this.TODOListItems.map((element) => {
+      if (id === element.id) {
+        return {
+          ...element,
+          done: !element.done
+        }
+      }
+      return element;
+    });
+    this.TODOListItems = newTodoItems;
+
+  }
+
+  @action
+  removeTodoItem = (itemId) => {
+    const newTodoItems = this.TODOListItems.filter((element) => element.id !== itemId);
+    this.TODOListItems = newTodoItems;
+  }
+
+  @action
+  addTodoItem = (e) => {
+    e.preventDefault();
+    const newTodo = {
+      "id": this.TODOListItems.length + 1,

```

Comment:
the Idea of `id` is to set a unique property for the object!
So the length in sommepoint won't be unique think about this:
>you have 3 items the length is 3 ids: 1, 2, 3
>you deleted a one in the middle so your ids now: 1, 3
>you added a new one your ids: 1, 3, 3

see? 🙃